xapian-core/include/xapian/cluster.h

```diff
@@ -219,8 +219,8 @@ class Centroid : public PointType {
     /// Constructor
     Centroid() { magnitude = 0; }
 
-    /// This method initializes the values of a centroid to the Point 'x'
-    void set_to_point(Point &x);
+    // Constructor with Point argument
+    Centroid(Point &x);
```

Comment:
It may feel intuitive to you to construct a `Centroid` with defaults and then set it to match a `Point`, but that feels very weird to me, and isn't a common pattern that I'm aware of. We aren't constructing centroids any other way, at the moment. (Which also means we probably want to make the default constructor — with no arguments — private.)

The _explicit_ constructor I'm fine with. Without the `explicit` keyword however it's an _implicit_ constructor, which means that you can do things like this:

```c++
void process_centroid(Centroid&);
Point p(…); // make a Point
process_centroid(p); // pass a Point to process_centroid
```

If the constructor were marked [`explicit`](http://en.cppreference.com/w/cpp/language/explicit), then you have to do:

```c++
process_centroid(Centroid(p));
```

which is clearer.